src/types/bloom_filter.h

```diff
@@ -46,6 +49,40 @@ constexpr bool IsMultipleOf8(int64_t n) { return (n & 7) == 0; }
 // This value will be reconsidered when implementing Bloom filter producer.
 static constexpr uint32_t kMaximumBloomFilterBytes = 128 * 1024 * 1024;
 
+/// Minimum Bloom filter size, it sets to 32 bytes to fit a tiny Bloom filter.
+static constexpr uint32_t kMinimumBloomFilterBytes = 32;
+
+class BlockSplitBloomFilter;
+
+using OwnedBlockSplitBloomFilter = std::tuple<BlockSplitBloomFilter, std::string>;
+
+/// Initialize the BlockSplitBloomFilter. The range of num_bytes should be within
+/// [kMinimumBloomFilterBytes, kMaximumBloomFilterBytes], it will be
+/// rounded up/down to lower/upper bound if num_bytes is out of range and also
+/// will be rounded up to a power of 2.
+///
+/// @param num_bytes The number of bytes to store Bloom filter bitset.
+OwnedBlockSplitBloomFilter CreateBlockSplitBloomFilter(uint32_t num_bytes);
+
+/// Initialize the BlockSplitBloomFilter. It copies the bitset as underlying
+/// bitset because the given bitset may not satisfy the 32-byte alignment requirement
```

Comment:
```suggestion
/// bitset when the given bitset may not satisfy the 32-byte alignment requirement
```

Would this be better?

Reply:
Oh, seems this bloom never copy, and when copy it's happening, it doesn't have owned-`std::string` as return value?

Reply:
it is just a view, does not own anything. If you want to a owned filter, you can have a std::tuple<BloomFilter, std::string>.